pydoctor/model.py

```diff
@@ -573,7 +573,7 @@ def setSourceHref(self, mod: _ModuleT, source_path: Path) -> None:
         if self.sourcebase is None:
             mod.sourceHref = None
         else:
-            projBaseDir = Path(mod.system.options.projectbasedirectory)
+            projBaseDir = mod.system.options.projectbasedirectory
```

Comment:
instead of validating the value at the high level config level, I would preferred to validate it at this lower level.

In the case in which we will pass a different options implementation, the value will still be validated.

At the same time, in this way we can keep the configuration simple with just a standard string value.

I am ok with having `projBaseDir` as a Path instance.
Is just that I prefer that the configuration not to be a path instance, but just a string instance.

Reply:
We don't actually validate the path at all; the name `check_path` is a bit misleading. I wasn't even able to trigger the exception handling without modifying the code to raise something, but I figured that if there is a string that `Path()` won't accept, we should report it without crashing.

If we would validate the path, I agree that it would be best to not do that in the config parser. However, I also think `setSourceHref()` is too deep and called too late. A better place could be `System.process()`.

I do like having `Path` instances in the config object though. I think it's similar to storing numeric values as `int`.